src/eko/io/exceptions.py

```diff
@@ -24,3 +24,35 @@ class OperatorLocationError(ValueError, OutputError):
     def __init__(self, path: os.PathLike):
         self.path = path
         super().__init__(f"Path '{path}' not in operators folder")
+
+
+class ReadOnlyOperator(RuntimeError, OutputError):
+    """It is not possible to write on a read-only operator.
+
+    In particular, the behavior would be deceitful, since writing is possible
```

Comment:
This should go into a comment rather than a docstring - it's only relevant for developers

Reply:
Nope, the error itself is relevant for whoever is getting the error. If the user will never get the error, so much the better, but in case, better if he can access an explanation as well. If he won't understand it, well, we loose nothing.